### PR TITLE
add null check around assembly & location

### PR DESCRIPTION
**Pull request by adamhathcock**
As found here: https://speckle.community/t/dllconflictmanager-throwing-error-in-revit/12589

### CODE REVIEW

File: ConnectorCore/DllConflictManagement/DllConflictManager.cs [hunk 1]
```diff
@@ -101,19 +101,37 @@ HashSet<string> visitedAssemblies
...
+  {
+    if (string.IsNullOrWhiteSpace(loadedAssembly?.Location))
+    {
+      return false;
+    }
+    string location = loadedAssembly!.Location;
+    if (!PathIsValid(location))
+    {
+      return false;
+    }
```
Comment (jsdbroughton) [inline ConnectorCore/DllConflictManagement/DllConflictManager.cs:123]: Do we want to log either of these exits?

Comment (adamhathcock) [reply ConnectorCore/DllConflictManagement/DllConflictManager.cs:123]: I don't want to log anything to our servers that doesn't matter.  We're logging too much as it is.

Comment (BovineOx) [reply ConnectorCore/DllConflictManagement/DllConflictManager.cs:123]: If there's an unhandled exception it should surface - we're not expecting one I guess

Comment (BovineOx) [reply ConnectorCore/DllConflictManagement/DllConflictManager.cs:123]: Might we want to check for NotSupportedException? https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=net-8.0

**Review by BovineOx — CHANGES_REQUESTED** [review]
I wonder if the NotSupportedException is a possibility here?

https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=net-8.0

File: ConnectorCore/DllConflictManagement/DllConflictManager.cs (hunk 1 above)
Comment (adamhathcock) [reply ConnectorCore/DllConflictManagement/DllConflictManager.cs:123]: maybe but it's likely already handled somewhere and it's not the current problem this is solving

**Review by JR-Morgan — APPROVED** [review]

**Pull request merged**